src/quicklist.c

```diff
@@ -924,9 +924,9 @@ REDIS_STATIC void _quicklistInsert(quicklist *quicklist, quicklistEntry *entry,
         new_node->count++;
         quicklistNodeUpdateSz(new_node);
         quicklistRecompressOnly(quicklist, new_node);
-    } else if (full && ((at_tail && node->next && full_next && after) ||
-                        (at_head && node->prev && full_prev && !after))) {
-        /* If we are: full, and our prev/next is full, then:
+    } else if (full && ((at_tail && full_next && after) ||
+                        (at_head && full_prev && !after))) {
+        /* If we are: full, and our prev/next is full or NULL, then:
```

Comment:
This optimization seems good to me. I think the `full_next` boolean variable is misleading because it actually means "full next or NULL next". So I'd change the logic above and call it `avail_next` which means next ziplist is not null and has available space. Then the fix will look like this:
```suggestion
    } else if (full && ((at_tail && !avail_next && after) ||
                        (at_head && !avail_prev && !after))) {
        /* If we are: full, and our prev/next has no available space, then:
```

Reply:
Agree.